cpp/benchmarks/ndsh/utils.cpp

```diff
@@ -430,28 +316,71 @@ int32_t days_since_epoch(int year, int month, int day)
   return static_cast<int32_t>(diff);
 }
 
-struct tpch_example_args {
-  std::string dataset_dir;
-  std::string memory_resource_type;
-};
-
-/**
- * @brief Parse command line arguments into a struct
- *
- * @param argc The number of command line arguments
- * @param argv The command line arguments
- */
-tpch_example_args parse_args(int argc, char const** argv)
+void write_to_parquet_device_buffer(std::unique_ptr<cudf::table> const& table,
+                                    std::vector<std::string> const& col_names,
+                                    parquet_device_buffer& source)
 {
-  if (argc < 3) {
-    std::string usage_message = "Usage: " + std::string(argv[0]) +
-                                " <dataset_dir> <memory_resource_type>\n The query result will be "
-                                "saved to a parquet file named q{query_no}.parquet in the current "
-                                "working directory ";
-    throw std::runtime_error(usage_message);
+  CUDF_FUNC_RANGE();
+  auto const stream = cudf::get_default_stream();
+
+  // Prepare the table metadata
+  cudf::io::table_metadata metadata;
+  std::vector<cudf::io::column_name_info> col_name_infos;
+  for (auto& col_name : col_names) {
+    col_name_infos.push_back(cudf::io::column_name_info(col_name));
   }
-  tpch_example_args args;
-  args.dataset_dir          = argv[1];
-  args.memory_resource_type = argv[2];
-  return args;
+  metadata.schema_info            = col_name_infos;
+  auto const table_input_metadata = cudf::io::table_input_metadata{metadata};
+
+  // Declare a host and device buffer
+  std::vector<char> h_buffer;
+
+  // Write parquet data to host buffer
+  auto builder =
+    cudf::io::parquet_writer_options::builder(cudf::io::sink_info(&h_buffer), table->view());
+  builder.metadata(table_input_metadata);
+  auto const options = builder.build();
+  cudf::io::write_parquet(options);
+
+  // Copy host buffer to device buffer
+  source.d_buffer.resize(h_buffer.size(), stream);
+  CUDF_CUDA_TRY(cudaMemcpyAsync(
```

Comment:
1. Looks like `cudf::io::source_info` does not work with `rmm::device_uvector<char>` and only works with `rmm::device_uvector<std::byte>` as it gives this error otherwise
```
/home/jayjeetc/project/cudf/cpp/benchmarks/ndsh/utilities.hpp:203:83: error: no matching function for call to ‘cudf::io::source_info::source_info(rmm::device_uvector<char>&)’
  203 |   cudf::io::source_info make_source_info() { return cudf::io::source_info(d_buffer); }
```

Reply:
2. Also, I can't make the `std::vector<char>` to `std::vector<std::byte>`, otherwise, I get
```
invalid conversion from ‘std::vector<std::byte>*’ to ‘size_t’ {aka ‘long unsigned int’} [-fpermissive]
  354 |     cudf::io::parquet_writer_options::builder(cudf::io::sink_info(&h_buffer), table->view());
      |                                                                   ^~~~~~~~~
      |                                                                   |
      |                                                                   std::vector<std::byte>*
```